src/main/java/xyz/nucleoid/codecs/MoreCodecs.java

```diff
@@ -187,7 +187,7 @@ public static <T> Codec<T> validate(Codec<T> codec, Function<T, DataResult<T>> v
         return codec.flatXmap(validate, validate);
     }
 
-    public static <T> Codec<T> validate(Codec<T> codec, Predicate<T> validate, String error) {
+    public static <T> Codec<T> validate(Codec<T> codec, Predicate<T> validate, Supplier<String> error) {
```

Comment:
I think we can deprecate this now, those exist in `Codecs` too 🙂

Reply:
Fixed in f7b2682f7a76e30ce1b4b61d4bf91f1a1aa0bee7